test/dashboard-e2e/pages/TestExecutionsPage.ts

```diff
@@ -1,16 +1,27 @@
-import type { Page } from  '@playwright/test';
+import {Page, expect} from '@playwright/test';
 
-export class TestExecutionsPage{
-    readonly page: Page;
-    constructor(page:Page){
-        this.page=page;
-    }
-    
-    async runTest() {
-        await this.page.click('//span[@class="ant-page-header-heading-extra"]//button[.//span]'); // TODO: data-test needed
-    }
-    
-    async openExecutionDetails(executionName) {
-        await this.page.click(`xpath=//tr[.//span[text()="${executionName}"]]`);
-    }
-}
\ No newline at end of file
+export class TestExecutionsPage {
+  readonly page: Page;
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async runTest() {
+    await this.page.click('//span[@class="ant-page-header-heading-extra"]//button[.//span]'); // TODO: data-test needed
+  }
+
+  async openExecutionDetails(executionName) {
+    await this.page.click(`xpath=//tr[.//span[text()="${executionName}"]]`);
+  }
+
+  async checkWebSocketOpened() {
+    const websocket = await this.page.waitForEvent('websocket');
+    const isClosed = await websocket.isClosed();
+    await expect(isClosed).toBeFalsy();
+  }
+
+  async checkExecutionLogs() {
+    const logs = await this.page.locator('//code');
+    await expect(logs).toContainText(`{"status":"running"}`);
```

Comment:
- `validateExecutionLogContents`? 
- expected text should be passed as argument

Should the logs really contain `{"status":"running"}`?

Reply:
- naming suggestion is great

first output of the logs stream is always something like `{"status":"running"}` . I would need to timeout the tests for more than 10-15 seconds to get more output from the stream. maybe by using k6 or postman tests, it will be faster. will check

Reply:
The `{"status":"running"}` is not necessarily there, for example, it doesn't appear for the finished test, and to be fair it's kind of a bug that it appears at all.

Maybe it would be better to add `async hasExecutionLogs(message: string)` instead, so it will be easier to use?

Reply:
agreed. I'll find a better message to assert against. As for the API, let's try to remove this message, since it's not relevant for the user

Reply:
I would consider as well `message: string | RegExp` - probably to detect logs it would be enough to check for `[^\t]+` 🙂